tools/salmon/salmonquant.xml

```diff
@@ -152,8 +152,78 @@
+                <param name="sampleUnaligned" value="--sampleUnaligned"/>
+                <param name="noErrorModel" value="--noErrorModel"/>
+                <param name="numErrorBins" value="5"/>
             </conditional>
-            <output name="postSample" file="postSample.bam" ftype="bam" />
+            <output name="postSample" file="postSample.bam" ftype="bam"/>
+        </test>
+        <test expect_num_outputs="1">
+            <conditional name="quant_type">
+                <param name="qtype" value="reads"/>
+                <conditional name="refTranscriptSource">
+                    <param name="TranscriptSource" value="history"/>
+                    <section name="s_index">
+                        <param name="fasta" value="transcripts.fasta"/>
+                    </section>
+                </conditional>
+                <section name="input">
+                    <param name="single_or_paired.single_or_paired_opts" value="paired" />
+                    <param name="single_or_paired.input_mate1" value="fastqs/reads_1.fastq" />
+                    <param name="single_or_paired.input_mate2" value="fastqs/reads_2.fastq" />
+                    <param name="single_or_paired.oreintation" value="U"/>
+                </section>
+                <param name="discardOrphansQuasi" value="-discardOrphansQuasi"/>
+                <conditional name="validmap">
+                    <param name="validateMappings" value="--validateMappings"/>
+                </conditional>
+                <param name="dovetail" value="--allowDovetail"/>
+                <param name="recoverOrphans" value="--recoverOrphans"/>
+            </conditional>
+            <output name="output_quant" ftype="tabular">
+                <assert_contents>
+                    <has_text text="EffectiveLength" />
+                    <has_text text="TPM" />
+                    <has_text text="NM_001168316" />
+                    <has_text text="NM_174914" />
+                    <has_text text="NM_018953" />
+                    <has_n_columns n="5" />
+                </assert_contents>
+            </output>
+        </test>
+        <test expect_num_outputs="1">
+            <conditional name="quant_type">
+                <param name="qtype" value="reads"/>
+                <conditional name="refTranscriptSource">
+                    <param name="TranscriptSource" value="history"/>
+                    <section name="s_index">
+                        <param name="fasta" value="transcripts.fasta"/>
+                    </section>
+                </conditional>
+                <section name="input">
+                    <param name="single_or_paired.single_or_paired_opts" value="paired" />
+                    <param name="single_or_paired.input_mate1" value="fastqs/reads_1.fastq" />
+                    <param name="single_or_paired.input_mate2" value="fastqs/reads_2.fastq" />
+                    <param name="single_or_paired.oreintation" value="U"/>
```

Comment:
```suppgestion
                    <param name="single_or_paired.orientation" value="U"/>
```

Reply:
The postSample.bam is showing up as the proper size with planemo test --no_cleanup, so it's pulling incorrectly. And I'll add in the loc file tests.

Reply:
Also fixed to strandedness